clients/da-vinci-client/src/main/java/com/linkedin/davinci/notifier/MetaSystemStoreReplicaStatusNotifier.java

```diff
@@ -1,12 +1,14 @@
 package com.linkedin.davinci.notifier;
 
+import com.linkedin.davinci.stats.AggHostLevelIngestionStats;
 import com.linkedin.venice.common.VeniceSystemStoreType;
 import com.linkedin.venice.meta.Instance;
 import com.linkedin.venice.meta.ReadOnlyStoreRepository;
 import com.linkedin.venice.meta.Store;
 import com.linkedin.venice.meta.Version;
 import com.linkedin.venice.pushmonitor.ExecutionStatus;
 import com.linkedin.venice.system.store.MetaStoreWriter;
+import org.apache.kafka.common.errors.TimeoutException;
```

Comment:
Hmm... that doesn't seem right. We should wrap Kafka-specific exceptions in our own exception type. We already have `VeniceOperationAgainstKafkaTimedOut` (which we should probably rename to PubSub rather than Kafka, but the rename doesn't necessarily have to happen as part of this commit...), can you use that one instead? That means the `ApacheKafkaProducerAdapter` and `ApacheKafkaProducerCallback` should do the exception wrapping.

FYI @sushantmane.

Reply:
> Thanks Min,
> 
> This change may be fine, but there are some things I am wondering about:
> 
> 1. Let's say the server is trying to use the `MetaStoreWriter` on one of the operations that have a flush with timeout, while Kafka has some availability issue and the flushing is hung, and then we shut down the server. Will the server's shut down be blocked by this or is the flushing going to be interrupted?
> 2. If the flushing is not going to be interrupted, does it mean the server in that scenario will wait up to 3 minutes extra before shutting down? Or can the 3 minutes flush happen several times in a row, leading to even longer shut down times?
> 
> Also, BTW, seems like there is a compilation failure in `clients:venice-push-job:compileTestJava`.

@FelixGV  Checked the implementation just now; closing `MetaStoreWriter` will close all cached VeniceWriter in parallel, and each of them has a 30 seconds timeout on flushing and closing (1 minute in total). They have a different config called "venice.writer.close.timeout.ms" (CLOSE_TIMEOUT_MS = VENICE_WRITER_CONFIG_PREFIX + "close.timeout.ms"). And server shutdown logic wouldn't interrupt this flushing.